catkin_tools/execution/executor.py

```diff
@@ -173,9 +173,9 @@ async def async_job(verb, job, threadpool, locks, event_queue, log_path):
             if retcode == stage.early_termination_retcode:
                 break
 
-            # Close logger
-            logger.close()
         finally:
+            logger.close()
```

Comment:
```suggestion
            if logger: logger.close()
```
?

`logger` is only assigned within the `try` block and may still be `None` if an exception is raised early.

Reply:
```suggestion
            if logger is not None:
                logger.close()
```